modules/text2vec-openai/clients/openai.go

```diff
@@ -243,6 +244,47 @@ func (v *client) getOpenAIOrganization(ctx context.Context) string {
 	return v.openAIOrganization
 }
 
+func (v *client) getRateLimit(ctx context.Context) (int, int) {
+	returnRPM := 0
+	returnTPM := 0
+	if rpmS := v.getValueFromContext(ctx, "X-Openai-Ratelimit-RequestPM-Embedding"); rpmS != "" {
+		s, err := strconv.Atoi(rpmS)
+		if err == nil {
+			returnRPM = s
+		}
+	}
+	if tpmS := v.getValueFromContext(ctx, "X-Openai-Ratelimit-TokenPM-Embedding"); tpmS != "" {
```

Comment:
header name needs to be in `X-Openai-Ratelimit-Tokenpm-Embedding` in order for this method to get the API Key from http request